server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/vacation/VacationResponseSet.scala

```diff
@@ -0,0 +1,62 @@
+package org.apache.james.jmap.vacation
+
+import java.time.ZonedDateTime
+
+import org.apache.james.jmap.api.vacation.VacationPatch
+import org.apache.james.jmap.model.AccountId
+import org.apache.james.jmap.model.State.State
+import play.api.libs.json.JsObject
+
+case class VacationResponseSetRequest(accountId: AccountId,
+                               update: Map[String, VacationResponsePatchObject]) {
+  def parsePatch(): Either[IllegalArgumentException, VacationResponsePatchObject] = {
+    update.map({
+      case (id, patch) => id match {
+        case "singleton" => Right(patch)
+        case _ => Left(new IllegalArgumentException("id must be singleton"))
+      }
+    }).head
+  }
+}
+
+case class VacationResponseId(value: String)
+
+case class IsEnabled(value: Boolean) extends AnyVal
+
+case class FromDate(value: ZonedDateTime)
+
+case class ToDate(value: ZonedDateTime)
+
+case class VacationResponsePatchObject(id: VacationResponseId,
+                                       isEnabled: IsEnabled,
+                                       fromDate: Option[FromDate],
+                                       toDate: Option[ToDate],
+                                       subject: Option[String],
+                                       textBody: Option[String],
+                                       htmlBody: Option[String]) {
```

Comment:
You have, in a patch to distinguish "absent" which means your property will be kept and "null" which means the property will be removed.

Your current VacationResponsePatchObject don't allow such a distinction.

Thus, I encourage you writing a test updating, for instance, just the textBody

Reply:
https://github.com/linagora/james-project/pull/3687/commits/d35e7bee2ae50a2c593b04c26905bbe05a8113c9